homeassistant/components/hive/__init__.py

```diff
@@ -69,75 +64,113 @@
 
         entity_lookup = hass.data[DOMAIN]["entity_lookup"]
-        hive_id = entity_lookup.get(service.data[ATTR_ENTITY_ID])
-        if not hive_id:
+        device = entity_lookup.get(service.data[ATTR_ENTITY_ID])
+        if not device:
             # log or raise error
             _LOGGER.error("Cannot boost entity id entered")
             return
 
         minutes = service.data[ATTR_TIME_PERIOD]
         temperature = service.data[ATTR_TEMPERATURE]
 
-        hive.heating.turn_boost_on(hive_id, minutes, temperature)
+        await hive.heating.turn_boost_on(device, minutes, temperature)
 
     async def hot_water_boost(service):
         """Handle the service call."""
         entity_lookup = hass.data[DOMAIN]["entity_lookup"]
-        hive_id = entity_lookup.get(service.data[ATTR_ENTITY_ID])
-        if not hive_id:
+        device = entity_lookup.get(service.data[ATTR_ENTITY_ID])
+        if not device:
             # log or raise error
             _LOGGER.error("Cannot boost entity id entered")
             return
+
         minutes = service.data[ATTR_TIME_PERIOD]
-        mode = service.data[ATTR_MODE]
+        mode = service.data[ATTR_ONOFF]
 
         if mode == "on":
-            hive.hotwater.turn_boost_on(hive_id, minutes)
+            await hive.hotwater.turn_boost_on(device, minutes)
         elif mode == "off":
-            hive.hotwater.turn_boost_off(hive_id)
+            await hive.hotwater.turn_boost_off(device)
 
-    hive = Hive()
+        # Update config entry options
 
-    config = {}
-    config["username"] = config[DOMAIN][CONF_USERNAME]
-    config["password"] = config[DOMAIN][CONF_PASSWORD]
-    config["update_interval"] = config[DOMAIN][CONF_SCAN_INTERVAL]
-
-    devices = await hive.session.startSession(config)
+    hive_options = hive_options if len(hive_options) > 0 else hive_config["options"]
+    hive_config["options"].update(hive_options)
+    hive_config["add_sensors"] = bool("hive" in hass.data["custom_components"])
+    hass.config_entries.async_update_entry(entry, options=hive_options)
+    hass.data[DOMAIN][entry.entry_id] = hive
+    hass.data[DOMAIN]["entity_lookup"] = {}
 
-    if devices is None:
-        _LOGGER.error("Hive API initialization failed")
-        return False
+    try:
+        hive.devices = await hive.session.startSession(hive_config)
+    except HTTPException as error:
+        _LOGGER.error("Could not connect to the internet: %s", error)
+        raise ConfigEntryNotReady() from error
 
-    hass.data[DOMAIN][DATA_HIVE] = hive
-    hass.data[DOMAIN]["entity_lookup"] = {}
+    if hive.devices == "INVALID_REAUTH":
+        return hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": config_entries.SOURCE_REAUTH}
+            )
+        )
 
-    for ha_type in DEVICETYPES:
-        devicelist = devices.get(DEVICETYPES[ha_type])
+    for component in PLATFORMS:
+        devicelist = hive.devices.get(component)
```

Comment:
There needs to be a decoupling between the hive names and the home assistant platform names. Please define a constant dict that maps these.

Reply:
The dictionary has keys the same as the home assistant platform names, so the dictionary would be.

```python
LOOKUP = {
"binary_sensor": "binary_sensor",
"climate": "climate",
"light": "light",
"sensor": "sensor",
"switch": "switch",
"water_heater": "water_heater"
}
```

This seems a but overkill to loop through a dictionary for the same value. Im happy to write a comment explaining this so others are aware.

Reply:
No we need this decoupling to make sure that we don't break silently if any of the two APIs change.